modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala

```diff
@@ -1071,6 +1080,50 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
       }
 
     errors.toList match {
+      case (h @ (_, _, e: Upload.Error.HttpError)) :: _
+          if isSonatype && errors.distinctBy(_._3.getMessage()).size == 1 =>
+        val httpCodeRegex = "HTTP (\\d+)\n.*".r
+        e.getMessage() match {
+          case httpCodeRegex("400") =>
+            logger.error(
+              s"""
+                 |Uploading files failed!
+                 |Possible causes:
+                 |- '-SNAPSHOT' suffix is missing or was added without need
+                 | -> we recommend setting the compute version as "git:tag"
+                 |- incorrect Sonatype server was used (legacy or s01)
+                 | -> consult publish subcommand documentation
+                 |- no rights to push under this organization or organization name is misspelled
+                 | -> have you registered your organisation yet?
+                 |""".stripMargin
+            )
+          case httpCodeRegex("403") =>
+            logger.error(
+              s"""
+                 |Uploading files failed!
+                 |Possible causes:
+                 |- no rights to push to under this organization or organization name is misspelled
+                 | -> have you registered your organisation yet?
+                 |- you're trying to update/republish already existing artifacts
+                 | -> be sure to increment your version with each non-snapshot upload
+                 |""".stripMargin
+            )
+          case _ => throw new UploadError(::(h, Nil))
+        }
+      case _ :: _ if isSonatype && errors.forall {
+            case (_, _, _: Upload.Error.Unauthorized) => true
+            case _                                    => false
+          } =>
+        logger.error(
+          s"""
+             |Uploading files failed!
+             |Possible causes:
+             |- incorrect Sonatype credentials
+             |- your Sonatype password or username may contain unsupported characters
```

Comment:
can we check earlier in our logic if any characters unsupported by `Sonatype` were used in the credentials?